### PR TITLE
upgrade to v2 communication data structure

### PR DESCRIPTION
**Pull request by shamasis**

### CODE REVIEW

File: lib/sandbox/postman-legacy-interface.js [hunk 1]
```diff
@@ -93,24 +187,151 @@ _.assign(PostmanLegacyTestInterface.prototype, /** PostmanLegacyTestInterface.pr
...
+         * The request that will be sent (or has been sent)
+         * @memberOf SandboxGlobals
+         * @type {Object}
+         */
+        globalvars.request = execution.request ? {
+            id: execution.legacy ? execution.legacy._itemId : undefined,
+            name: execution.legacy ? execution.legacy._itemName : undefined,
+            description: execution.request.description ?
+                execution.request.description.toString() : undefined,
+            headers: execution.request.getHeaders(),
```
Comment (shamasis) [inline lib/sandbox/postman-legacy-interface.js]: @czardoz sos

File: lib/sandbox/postman-legacy-interface.js [hunk 2]
```diff
@@ -73,16 +171,12 @@ _.assign(PostmanLegacyInterface.prototype, /** PostmanLegacyInterface.prototype
...
-    this.responseHeaders = _.mapKeys(_.get(options, 'globals.responseHeaders', {}), function (val, key) {
-        return key.toLowerCase();
-    });
-
-    this.responseCookies = {};
-    _.each(_.get(options, 'globals.responseCookies', []), function (cookie) {
-        _.isString(cookie && cookie.name) && (this.responseCookies[cookie.name.toLowerCase()] = cookie);
+    this.__cookies = {};
+    _.each(execution.cookies, function (cookie) {
+        _.isString(cookie && cookie.name) && (this.__cookies[cookie.name.toLowerCase()] = cookie);
```
Comment (shamasis) [inline lib/sandbox/postman-legacy-interface.js]: @czardoz - sos

File: lib/sandbox/postman-legacy-interface.js (hunk 1 above)
Comment (czardoz) [inline lib/sandbox/postman-legacy-interface.js]: Looks like pathVariables are not exposed in the sandbox, so we're good. From the Chrome App:

image

File: lib/sandbox/postman-legacy-interface.js (hunk 2 above)
Comment (czardoz) [inline lib/sandbox/postman-legacy-interface.js]: Okay, I can confirm that this is supposed to be lowercased

**Pull request merged**